### DISCUSSION TITLE
Craft vs buying cost difference too variable for ammunition

### DISCUSSION THREAD

**Discussion by FourGreenFields** [FakeFishGames/Barotrauma#12753] [Unstable feedback]
TL;DR: The percentage of marks you save by crafting your ammunition should probably be a little more consistent between ammo-types.

Did the maths using the base price of ammunition and materials (i.e. not including discounts for buying from mining or military outposts). Some types of ammunition cost as little as a quarter of their purchasing cost when crafted, while for others you barely save any marks at all. Using expensive materials it's often _more_ expensive to craft than to buy.

What doesn't help with balancing is the price-difference even one ammo-type can have, depending on which materials you use for the munition parts. "Fixing" that would probably require changing the prices of some materials... which in turn would require balance-changes for other items too. I do hope this can be improved, but I feel this is a secondary issue compared to assuming the cheapest material-choices.

The difference is currently great enough that revolver rounds are the second most crafting-cost-efficient gun ammo (after SMG mag fired from an SMG), but are towards the lower end of cost-efficiency when purchased directly. Meanwhile the HMG is "quite normal" when crafting ammo, but incredibly cost-efficient when buying instead (on that note, I remain of the opinion that [the HMG needs another look at](https://github.com/Regalis11/Barotrauma/discussions/10773)).

Green highlights cases where you save over half the buying cost. Yellow is where you save less than 20%. Red is where buying would be cheaper. (Also note that depleted SMG mags being more expensive to craft doesn't matter, because they can't be bought)
![image](https://github.com/Regalis11/Barotrauma/assets/78027300/3fbe2328-9499-4278-9eba-c61f1a58ae1d)
The two on the left compare price if crafting from the cheapest option for munition parts. The two on the right for the most expensive option. The cost of the magazine itself (generally steel; didn't maths it for fuel rods) doesn't really make a difference.

Maths: [the "Ranged Handheld" sheet here.](https://docs.google.com/spreadsheets/d/1Vp-PO6pLo-ks_hYEShmsBXoic-XTnsfj58H896HuDPo/edit?usp=sharing)

**Comment by Ek-Videogames** [comment]
Ammo has always been exceedingly goofy. +1

Add turret ammo too.
